### PR TITLE
Final pull request for set game path notification

### PR DESCRIPTION
**Pull request by wehigami**
Whenever the game path is not set, a notification shows up that reminds you to do that. Hopefully there won't be conflicts anymore.

### CODE REVIEW

**Comment by SpikeHD** [comment]
I don't have a chance to test this at the moment, but this seems like it will notify every time the launcher is reopened, correct? Even if they have changed it in a previous session? It also disregards whether the default path is actually already correct or not (since the default path that is set upon first install is the default install for most people)

**Comment by wehigami** [comment]
> I don't have a chance to test this at the moment, but this seems like it will notify every time the launcher is reopened, correct? Even if they have changed it in a previous session? It also disregards whether the default path is actually already correct or not (since the default path that is set upon first install is the default install for most people)
Nope, from my tests it will not notify every time you open the launcher and it also does not disregard the default path. 
If there's no path, it notifies.

**Comment by NotThorny** [comment]
Works as described, but since the default path is set for every user, the only time this will notify is if the user manually removes their current path in settings. Would be nice if it checked default path existing, like Spike said above, and then also notifies in that case.
The average user with a different install will see the game is "set" even though the actual exe doesn't exist for them there, and still not know why the game isn't launching. They aren't seeing the notification since they have a game path set, just not a valid one.

Might also be a good idea to replace the span text with a tr so that other languages can have the right message there.

File: src/ui/Main.tsx [hunk 1]
```diff
@@ -185,6 +195,21 @@ export class Main extends React.Component<IProps, IState> {
     })
   }
 
+  async componentDidUpdate(prevProps: Readonly<IProps>, prevState: Readonly<IState>) {
+    const game_path = await getConfigOption('game_install_path')
+
```
Comment (NotThorny) [inline src/ui/Main.tsx:200]:
```suggestion

    // Check if game exists at set location
    const game_exists: boolean = ((await invoke('dir_exists', {
      path: game_path,
    })) as boolean)

    // Set no game path so the user understands it doesn't exist there
    if (!game_exists) {
      this.setState({
        game_install_path: '',
      })
    }

```

**Comment by wehigami** [comment]
@NotThorny 
Oh, I didn't understand what spike meant. This seems like a great idea.
Also I wouldn't even have that such a solution may exist. Thank you!

**Comment by NotThorny** [comment]
Sure thing, just updated it to work properly since I hadn't actually tested it when writing it the first time. So if the default path doesn't exist for the user, it will unset the game path and alert that one needs to be set.

**Comment by wehigami** [comment]
@NotThorny 
Do I just push the change or create a new request?

**Comment by NotThorny** [comment]
> @NotThorny Do I just push the change or create a new request?

You can just commit the suggested changes in this branch/PR, no need to make a new one

**Comment by wehigami** [comment]
Commited the changes, I think this should be working now

**Comment by SpikeHD** [comment]
Prettier checks fail, mind running that and committing the changes pls?

File: src/ui/Main.tsx [hunk 2]
```diff
@@ -185,6 +195,31 @@ export class Main extends React.Component<IProps, IState> {
...
+    // Check if game exists at set location
+    const game_exists: boolean = ((await invoke('dir_exists', {
```
Comment (NotThorny) [inline src/ui/Main.tsx:202]:
```suggestion
    const game_exists: boolean = (await invoke('dir_exists', {
      path: game_path,
    })) as boolean
```

File: src/ui/Main.tsx [hunk 3]
```diff
@@ -59,6 +62,8 @@ export class Main extends React.Component<IProps, IState> {
         alert('Error launching game')
       },
       notification: null,
+      isGamePathSet: false,
```
Comment (NotThorny) [inline src/ui/Main.tsx:65]:
```suggestion
      isGamePathSet: true,
```

File: src/ui/Main.tsx (hunk 2 above)
Comment (NotThorny) [reply src/ui/Main.tsx:202]: This should be the only prettier change

File: src/ui/Main.tsx (hunk 3 above)
Comment (NotThorny) [reply src/ui/Main.tsx:65]: This prevents notification every time launcher is opened, like Spike mentioned before. Notification will still properly occur when path is not set or incorrect, but will stop it from flashing on open when the path is set correctly.

**Pull request merged**